src/transformers/models/align/processing_align.py

```diff
@@ -16,8 +16,60 @@
 Image/Text processor class for ALIGN
 """
 
-from ...processing_utils import ProcessorMixin
-from ...tokenization_utils_base import BatchEncoding
+from typing import List, Union
+
+
+try:
+    from typing import Unpack
+except ImportError:
+    from typing_extensions import Unpack
+
+from ...image_utils import ImageInput
+from ...processing_utils import (
+    CommonKwargs,
+    ImagesKwargs,
+    ProcessingKwargs,
+    ProcessorMixin,
+    TextKwargs,
+    is_vision_available,
+)
+from ...tokenization_utils_base import BatchEncoding, PreTokenizedInput, TextInput
+from ...utils import is_torch_available
+
+
+# TODO (@molbap) This is a bother, forward references from TypedDict are resolved and need this to work
+if is_vision_available():
+    import PIL  # noqa: F401
+if is_torch_available():
+    import torch  # noqa: F401
+
+
+class AlignProcessorKwargs(ProcessingKwargs, CommonKwargs, TextKwargs, ImagesKwargs, total=False):
+    """
+    Inherits from `ProcessingKwargs` to provide:
+        1) Additional keys that this model requires to process inputs.
+        2) Default values for extra keys.
+    New keys have to be defined as follows to ensure type hinting is done correctly.
+
+    ```python
+    images_kwargs: ImagesKwargs = {"new_image_kwarg": Optional[bool]}
+
+    _defaults = {
+        "text_kwargs": {
+            "padding": "max_length",
+            "max_length": 64,
+        },
+    }
+
+    ```
+    """
+
```

Comment:
```suggestion
```

Reply:
I think the documentation should exist somewhere though, what about in `processing_utils`?

Reply:
that sounds good 😉

Reply:
done!